core/src/main/java/org/apache/accumulo/core/trace/TraceUtil.java

```diff
@@ -191,22 +194,13 @@ public static <T> Callable<T> unwrap(Callable<T> c) {
 
   public static <T> T wrapService(final T instance) {
     InvocationHandler handler = (obj, method, args) -> {
-      if (args == null || args.length < 1 || args[0] == null || !(args[0] instanceof TInfo)) {
-        try {
-          return method.invoke(instance, args);
-        } catch (InvocationTargetException e) {
-          throw e.getCause();
-        }
-      }
-      Span span = startServerRpcSpan(instance.getClass(), method.getName(), (TInfo) args[0]);
-      try (Scope scope = span.makeCurrent()) {
+      Span span = Span.current(); // should be set by protocol
+      try {
```

Comment:
So, I think that what the old code was doing when it called `startServerRpcSpan` was that it was always creating a new "inner span" inside the outer span that that the user might have created in the client. That was useful because it helped us distinguish the a child or inner span that was just associated with the RPC call, rather than be grouped with all the other things the client code might have been doing around the RPC call.

This code, however, seems to just grab the current span the user created, and attach an exception to it if an RPC exception occurred. I think we should do the other thing. If there's an exception, it should be attached to the inner/child span for the RPC, not the outer span the user created.

Reply:
Okay I think I fixed this in 8d742ea. Please take a look.

Reply:
I think I may have misunderstood what you were doing before. I think you might have had it correct before. I didn't realize you were creating the server span in the protocol, prior to this invocation handler being called. The expectation, I think, was that the invocation handler just needed to get a reference to the span started by the protocol, for the purposes of attaching any exceptions to it. I think I will need to look at this more closely, but I am now thinking I misled you by my earlier comment, and you were doing it correctly before.